extensions/amp-story/1.0/amp-story.js

```diff
@@ -794,7 +795,13 @@ export class AmpStory extends AMP.BaseElement {
             });
           }
         })
-        .then(() => this.switchTo_(initialPageId, NavigationDirection.NEXT))
+        .then(() => {
+          let maybeInitialPageId = initialPageId;
+          if (isExperimentOn(this.win, 'amp-story-branching')) {
+            maybeInitialPageId = this.maybeChangeInitial_(initialPageId);
+          }
+          this.switchTo_(maybeInitialPageId, NavigationDirection.NEXT);
+        })
```

Comment:
Chaining promises is pretty confusing, but here you're actually changing the way this code behave.

The `this.switchTo_()` method returns a promise, that resolves when the page navigation happened.

The way the code was written was using an arrow function, with an implicit return
`() => 'foo'` is the shorthand for `() => {return 'foo';}`
What it means is that the way it was written before, it was returning the Promise from the switchTo_ method, and the next `then()` callback wasn't executed until the Promise was resolved (navigation happens).
Your new code doesn't return that Promise, which means that the following code will run before it's actually resolved.

Reply:
Thank you for catching this and for taking the time to explain everything out! I made changes, but will definitely keep in mind the implicit return and be more perceptive about chained promises in the future.